sdk/eventhub/event-hubs/src/eventHubReceiver.ts

```diff
@@ -547,7 +563,82 @@ export class EventHubReceiver extends LinkEntity {
     return result;
   }
 
-  protected _deleteFromCache(): void {
+  /**
+   * Registers the user's onMessage and onError handlers.
+   * Sends buffered events from the queue before adding additional credits to the AMQP link.
+   * @param onMessage
+   * @param onError
+   * @param abortSignal
+   * @ignore
+   */
+  registerHandlers(
+    onMessage: OnMessage,
+    onError: OnError,
+    maximumCreditCount: number,
+    isStreaming: boolean = false,
+    abortSignal?: AbortSignalLike
+  ): void {
+    this._onError = onError;
+    this._onMessage = onMessage;
+
+    // indicate that messages are being received.
+    this._isReceivingMessages = true;
+
+    this._useInternalQueue(onMessage, abortSignal)
+      .then(async (processedEventCount) => {
+        if (this._onMessage !== onMessage) {
+          // the original handler has been removed, so no further action required.
+          return;
+        }
+
+        // check if more messages are required
+        if (!isStreaming && maximumCreditCount - processedEventCount <= 0) {
+          return;
+        }
+
+        // ensure we creat an amqpReceiver with no credit window
+        this.prefetchCount = 0;
```

Comment:
Looks like we are always creating the receiver with no credit window.
If that is the case, then why not set the credit_window to 0 directly?

Reply:
Good point!